app/src/main/java/com/example/plannr/students/TableMaker.java

```diff
@@ -55,7 +55,7 @@ public void getWhatTake(CourseHash toBe, ArrayList<CourseHash> available) {
      */
     public ArrayList<String> buildTable(int year) {
         //initialize all needed variables
-        ArrayList<String> taken = new ArrayList<>(StudentUserModel.getInstance().getTakenCourses());
+        ArrayList<String> taken = new ArrayList<>(Course.stringToArraylist(StudentUserModel.getInstance().getTakenCourses()));
```

Comment:
Course.stringToArraylist i bellieve splits by "," so be careful of that
Also i believe when you split, you end up with an empty string "" so just in case just do a taken.remove(""); in the next line

Reply:
in my most recent commit, instead of splitting the string I just take the already split courses list object